com.vrcfury.vrcfury/Editor/VF/Feature/CombineMeshesBuilder.cs

```diff
@@ -0,0 +1,543 @@
+            content.Add(new VisualElement { style = { paddingTop = 10 } });
+            return content;
+        }
+
+        public override bool AvailableOnProps()
+        {
+            return false;
+        }
+
+        [FeatureBuilderAction(FeatureOrder.BlendshapeOptimizer)]
+        public void Apply()
+        {
+            var changes = CalculateMeshChanges();
+            foreach (var group in changes.Groups)
+            {
+                var skins = group.Value;
+                var basis = skins[0];
+                VFGameObject basisObject = basis.gameObject;
+                var meshesToAdd = skins.Skip(1).ToList();
+                basis.sharedMesh = mutableManager.MakeMutable(basis.sharedMesh, basis.owner()); ;
+                VRCFuryEditorUtils.MarkDirty(basis);
+
+                var combinable = new CombinableMesh(basis);
+                foreach (var mesh in meshesToAdd)
+                {
+                    combinable.AddMesh(mesh);
+                    VFGameObject obj = mesh.gameObject;
+                    mover.DirectRewrite(obj, basisObject);
+                    obj.Destroy();
+                }
+                combinable.Combine();
+            }
+            foreach (var skin in changes.Removed)
+            {
+                VFGameObject obj = skin.gameObject;
+                obj.Destroy();
+            }
+        }
+
+        private struct MeshChanges
+        {
+            public Dictionary<SkinnedMeshRendererGroup, List<SkinnedMeshRenderer>> Groups { get; set; }
+            public List<SkinnedMeshRenderer> Toggled { get; set; }
+            public List<SkinnedMeshRenderer> Removed { get; set; }
+            public List<SkinnedMeshRenderer> NotCombinable { get; set; }
+        }
+
+        private MeshChanges CalculateMeshChanges()
+        {
+            var task = new MeshChanges();
+            task.Groups = new Dictionary<SkinnedMeshRendererGroup, List<SkinnedMeshRenderer>>();
+            task.Removed = new List<SkinnedMeshRenderer>();
+            task.Toggled = new List<SkinnedMeshRenderer>();
+            task.NotCombinable = new List<SkinnedMeshRenderer>();
+
+            var descriptor = avatarObject.GetComponent<VRC_AvatarDescriptor>();
+            var faceMesh = descriptor.VisemeSkinnedMesh;
+
+            var animatedBindings = manager.GetAllUsedControllersRaw()
+                      .Select(tuple => tuple.Item2)
+                      .SelectMany(controller => GetBindings(avatarObject, controller))
+                      .Concat(avatarObject.GetComponentsInSelfAndChildren<Animator>()
+                          .SelectMany(animator => GetBindings(animator.gameObject, animator.runtimeAnimatorController as AnimatorController)))
+                      .Where(a =>
+                      {
+                          var (binding, _) = a;
+                          return binding.type == typeof(GameObject) && binding.propertyName == "m_IsActive";
```

Comment:
This needs to disqualify a lot more things, including:
* Is any property of the renderer animated, including things like blendshapes (which may be merged), material slots, etc
* Is the renderer itself disabled
* Is any parent object of the renderer disabled

Reply:
I added checks for the things you mentioned, can you think of anything else that we should check?